src/ray/raylet/agent_manager.cc

```diff
@@ -163,19 +141,20 @@ void AgentManager::CreateRuntimeEnv(
   }
 
   if (runtime_env_agent_client_ == nullptr) {
-    // If the agent cannot be restarted anymore, fail the request.
-    if (agent_restart_count_ >= RayConfig::instance().agent_max_restart_count()) {
+    // If the grpc service of agent is invalid, fail the request.
+    if (disable_agent_client_) {
       std::stringstream str_stream;
-      str_stream << "Runtime environment " << serialized_runtime_env
-                 << " cannot be created on this node because the agent is dead.";
+      str_stream
+          << "Runtime environment " << serialized_runtime_env
+          << " cannot be created on this node because the agent client is disabled. You "
+             "see this error message maybe because the grpc port of agent came into "
+             "conflict. Please see `dashboard_agent.log` to get more details.";
```

Comment:
```suggestion
          << "The runtime environment " << serialized_runtime_env
          << " cannot be created on this node because the agent client is disabled. You "
             "might see this error message because the gRPC port of the agent came into "
             "conflict. Please see `dashboard_agent.log` to get more details.";
```